pkg/k8s/apis/cilium.io/v2alpha1/cnc_types.go

```diff
@@ -12,7 +12,6 @@ import (
 //+kubebuilder:resource:categories={cilium}
 //+kubebuilder:object:root=true
 //+deepequal-gen=false
-//+kubebuilder:storageversion
```

Comment:
We'll want to add a deprecation marker here as well, https://github.com/cilium/controller-tools/blob/v0.9/pkg/crd/markers/crd.go#L52. I think it's important to also call out the versions of Cilium in which the deprecation will happen (the upcoming one) and when the deprecated version will be removed (the upcoming one + 1). 

See https://github.com/openyurtio/openyurt/blob/56b874cc7b153233199b9dcdf9bb49b116a77a5e/pkg/apis/iot/v1alpha1/platformadmin_types.go#L112 as an example.

Reply:
I added it and tested that the warning is visible when applying a CNC v2alpha1.